File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.php
// Last updated: 8/7/2025, 6:18:35 PM
class Solution {

    /**
     * @param Integer[] $nums
     * @return Integer
     */
    function findMaxConsecutiveOnes($nums) {
        $max = 0;
        $k = 0;
        for ($i = 0; $i<count($nums); $i++) {
            if ($nums[$i] == 1) {
                $max = max($i - $k + 1, $max);
            } else {
                $k = $i+1;
            }
        }
        return $max;
    }
}

File: 485-MaxConsecutiveOnes/485-MaxConsecutiveOnes.java
// Last updated: 8/7/2025, 6:18:38 PM
class Solution {
    public int findMaxConsecutiveOnes(int[] nums) {
        int max = 0;
        int result = 0;

        for (int i=0; i<nums.length; i++) {
            
            if (nums[i] == 1) {
                result++;
            } else {
                max = max <= result ? result : max;
                result = 0;
            }
        }
        return max = max <= result ? result : max;
    }
}